webapp/views/App/views/Designer/CategoryList/TableRow/store/actions/useEdit.js

```diff
@@ -15,13 +15,18 @@ export const useEdit = () => {
   const history = useHistory()
 
   return useCallback(({ state }) => {
-    const category = State.getCategory(state)
     const inCategoriesPath = State.isInCategoriesPath(state)
+    const onEdit = State.getOnEdit(state)
+    const category = State.getCategory(state)
     const categoryUuid = Category.getUuid(category)
+
     if (inCategoriesPath) {
       history.push(`${appModuleUri(designerModules.category)}${categoryUuid}`)
     } else {
       dispatch(CategoryActions.setCategoryForEdit(categoryUuid))
     }
+    if (onEdit) {
+      onEdit(category)
+    }
```

Comment:
can this just be if(onEdit) onEdit() else history.push() ?

Reply:
uhm, yes, you are right